### ISSUE TITLE
Dropdown remains static on iPhone app

### ISSUE THREAD

**Issue by cnickel76** [AhmedLSayed9/dropdown_button2#255]
I'm using the multi select dropdown widget in a Flutterflow app, within a drawer (drawn from the left hand side of the page). It works fine on Android and iPad, but not on iphone. There the widget does not unfold, it stays static and useless. Can someone help? Thanks

**Comment by AhmedLSayed9** [comment]
Please make a minimal reproducible example.

**Comment by cnickel76** [comment]
![Capture d'écran 2024-04-15 161419](https://github.com/AhmedLSayed9/dropdown_button2/assets/77248659/d66e5cfc-fb94-44cd-b886-fe8e29bb7b48)

**Comment by cnickel76** [comment]
this is the code of my custom widget in Flutterflow: 
// Automatic FlutterFlow imports
import '/backend/backend.dart';
import '/backend/schema/structs/index.dart';
import '/backend/supabase/supabase.dart';
import '/flutter_flow/flutter_flow_theme.dart';
import '/flutter_flow/flutter_flow_util.dart';
import '/custom_code/widgets/index.dart'; // Imports other custom widgets
import '/custom_code/actions/index.dart'; // Imports custom actions
import '/flutter_flow/custom_functions.dart'; // Imports custom functions
import 'package:flutter/material.dart';
// Begin custom widget code
// DO NOT REMOVE OR MODIFY THE CODE ABOVE!

import 'package:dropdown_button2/dropdown_button2.dart';

// ignore: must_be_immutable
class MyDDWidget2 extends StatefulWidget {
  @override
  MyDDWidget2({
    super.key,
    this.width,
    this.height,
    required this.items,
    required this.selectedItems,
  });

  final double? width;
  final double? height;
  List<String> items;
  List<String> selectedItems;
  @override
  State<MyDDWidget2> createState() => _MyDDWidget2State();
}

class _MyDDWidget2State extends State<MyDDWidget2> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: DropdownButtonHideUnderline(
          child: DropdownButton2<String>(
            isExpanded: true,
            hint: Text(
              'Select Category',
              style: TextStyle(
                fontSize: 14,
                color: Theme.of(context).hintColor,
              ),
            ),
            items: widget.items.map((item) {
              return DropdownMenuItem(
                value: item,
                //disable default onTap to avoid closing menu when selecting an item
                enabled: false,
                child: StatefulBuilder(
                  builder: (context, menuSetState) {
                    final isSelected = widget.selectedItems.contains(item);
                    return InkWell(
                      onTap: () {
                        isSelected
                            ? widget.selectedItems.remove(item)
                            : widget.selectedItems.add(item);
                        //This rebuilds the StatefulWidget to update the button's text
                        FFAppState().update(() {
                          FFAppState().selEventCats = widget.selectedItems;
                        });
                        //This rebuilds the dropdownMenu Widget to update the check mark
                        menuSetState(() {});
                      },
                      child: Container(
                        height: double.infinity,
                        padding: const EdgeInsets.symmetric(horizontal: 16.0),
                        child: Row(
                          children: [
                            if (isSelected)
                              const Icon(Icons.check_box_outlined)
                            else
                              const Icon(Icons.check_box_outline_blank),
                            const SizedBox(width: 16),
                            Expanded(
                              child: Text(
                                item,
                                style: const TextStyle(
                                  fontSize: 12,
                                ),
                              ),
                            ),
                          ],
                        ),
                      ),
                    );
                  },
                ),
              );
            }).toList(),
            //Use last selected item as the current value so if we've limited menu height, it scroll to last item.
            value:
                widget.selectedItems.isEmpty ? null : widget.selectedItems.last,
            onChanged: (value) {},
            selectedItemBuilder: (context) {
              return widget.items.map(
                (item) {
                  return Container(
                    alignment: AlignmentDirectional.center,
                    child: Text(
                      widget.selectedItems.join(', '),
                      style: const TextStyle(
                        fontSize: 12,
                        overflow: TextOverflow.ellipsis,
                      ),
                      maxLines: 1,
                    ),
                  );
                },
              ).toList();
            },
            buttonStyleData: const ButtonStyleData(
              padding: EdgeInsets.only(left: 16, right: 8),
              height: 30,
              width: 180,
            ),
            menuItemStyleData: const MenuItemStyleData(
              height: 30,
              padding: EdgeInsets.zero,
            ),
          ),
        ),
      ),
    );
  }
}

**Comment by AhmedLSayed9** [comment]
Unfortunately, I can't help with that. I can't run the code given.

I'll need a minimal reproducible example.

**Comment by cnickel76** [comment]
Well I do not know what to add here... the code that I pasted was simply taken from your pub.dev package webpage.

**Comment by AhmedLSayed9** [comment]
Yes, but I can't run that sample. It has dependencies that I don't have:
```dart
import '/backend/backend.dart';
import '/backend/schema/structs/index.dart';
import '/backend/supabase/supabase.dart';
import '/flutter_flow/flutter_flow_theme.dart';
import '/flutter_flow/flutter_flow_util.dart';
import '/custom_code/widgets/index.dart'; // Imports other custom widgets
import '/custom_code/actions/index.dart'; // Imports custom actions
import '/flutter_flow/custom_functions.dart'; // Imports custom functions
```

**Comment by AhmedLSayed9** [comment]
Without a runnable sample I can't help with this. Feel free to ask for re-open if you have further information.
